include/odp/api/spec/packet_io.h

```diff
@@ -1394,6 +1397,18 @@ uint64_t odp_pktin_ts_res(odp_pktio_t pktio);
  */
 odp_time_t odp_pktin_ts_from_ns(odp_pktio_t pktio, uint64_t ns);
 
+/**
+ * Get Tx timestamp inserted for a packet if available and clear it for
+ * next timestamp.
+ *
+ * @param pktio handle identifying pktio
+ * @param ts    Pointer to odp_time_t where timestamp will be written
+ *
+ * @retval zero     Successfully returns last pkt timestamp.
+ * @retval negative error code on failure.
```

Comment:
We probably should distinguish between different kind of errors: 1) Time stamp not available because none has been requested, 2) Time stamp not yet available for a sent packet, try again later, 3) Time stamping failed or other error occured, retrying is useless.

Maybe 1) and 2) could be the same error code.

Reply:
Ok. Will fix it with distinguishing between 1), 2) error and 3) error.